File: src/main/java/br/com/correios/enderecador/util/Cep.java
package br.com.correios.enderecador.util;

public class Cep {
    public static boolean validaCep(String cep) {
        StringBuilder novoCep = new StringBuilder();
        boolean retorno = true;
        if (cep.length() < 8) {
            retorno = false;
        } else {
            for (int i = 0; i < cep.length(); i++) {
                char c = cep.charAt(i);
                if (Character.isDigit(c))
                    novoCep.append(c);
            }
            if (novoCep.length() != 8)
                retorno = false;
        }
        return retorno;
    }

    public static String tiraMascaraCep(String cep) {
        StringBuilder novoCep = new StringBuilder();
        for (int i = 0; i < cep.length(); i++) {
            char c = cep.charAt(i);
            if (Character.isDigit(c))
                novoCep.append(c);
        }
        if (novoCep.length() == 8)
            return novoCep.toString();
        return "";
    }
}

File: src/main/kotlin/br/com/correios/enderecador/util/Cep.kt
package br.com.correios.enderecador.util

import java.lang.StringBuilder

object Cep {
    fun validaCep(cep: String): Boolean {
        val novoCep = StringBuilder()
        var retorno = true
        if (cep.length < 8) {
            retorno = false
        } else {
            for (element in cep) {
                if (Character.isDigit(element)) {
                    novoCep.append(element)
                }
            }
            if (novoCep.length != 8) {
                retorno = false
            }
        }
        return retorno
    }

    fun tiraMascaraCep(cep: String): String {
        val novoCep = StringBuilder()
        for (element in cep) {
            if (Character.isDigit(element)) {
                novoCep.append(element)
            }
        }
        return if (novoCep.length == 8) {
            novoCep.toString()
        } else ""
    }
}
